core/src/main/java/com/linecorp/armeria/server/RoutingStatus.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A routing status of an incoming HTTP request.
+ */
+@UnstableApi
+public enum RoutingStatus {
+    /**
+     * The routing completed successfully.
+     */
+    OK,
+    /**
+     * An {@code "OPTIONS * HTTP/1.1"} request.
+     */
+    OPTIONS,
+    /**
+     * The request specified an invalid path.
+     */
+    INVALID_PATH
```

Comment:
Can we also add `CORS_PREFLIGHT` and replace `RoutingContext.isCorsPreflight()`?